service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/utils/TaskStatusChecker.java

```diff
@@ -49,33 +56,25 @@ public static synchronized void init(CassandraConnectionProvider cassandraConnec
 
 
     public boolean hasKillFlag(long taskId) {
-        CacheItem cacheItem = getCacheItem(taskId);
-        if (!cacheItem.hasKilledFlag && cacheItem.lastCheck + CHECKING_INTERVAL < System.currentTimeMillis()) {
-            synchronized (instance) {
-                if (cacheItem.lastCheck + CHECKING_INTERVAL < System.currentTimeMillis()) {
-                    LOGGER.info("Checking the cancellation from the backend database for task id {}", taskId);
-                    cacheItem.lastCheck = System.currentTimeMillis();
-                    if (taskDAO.hasKillFlag(taskId)) {
-                        cacheItem.hasKilledFlag = true;
-                        cache.put(taskId, cacheItem);
-                    }
-                }
-            }
-
+        try {
+            return cache.get(taskId);
+        } catch (ExecutionException e) {
+            LOGGER.info(e.getMessage());
+            return false;
         }
-        return cacheItem.hasKilledFlag;
     }
 
-
-    private CacheItem getCacheItem(long taskId) {
-        ConcurrentMap<Long, CacheItem> map = cache.asMap();
-        map.putIfAbsent(taskId, new CacheItem());
-        return map.get(taskId);
-    }
-
-    private static class CacheItem {
-        volatile boolean hasKilledFlag = false;
-        volatile long lastCheck = 0;
+    /*
+       This method will only be executed if there is no VALUE for KEY taskId inside cache or if refresh method was triggered.
+       In the current implementation it will be triggered every 5 seconds if it was queried.
+     */
+    private Boolean isTaskKilled(long taskId) {
+        boolean isKilled = false;
+        LOGGER.info("Checking the task status for the task id from backend: " + taskId);
```

Comment:
Use the internal formatting? Like so:
LOGGER.info("Checking the task status for the task id from backend: {}", taskId);

Reply:
True. Done!